core/src/main/java/io/fabric8/maven/core/service/openshift/ImageStreamService.java

```diff
@@ -205,29 +208,59 @@ private String findTagSha(OpenShiftClient client, String imageStreamName, String
             if (tags == null || tags.isEmpty()) {
                 continue;
             }
-            // latest tag is the first
+
+            // Iterate overall tags and get the latest one by 'created' attribute
+            TagEvent latestTag = null;
+
             TAG_EVENT_LIST:
             for (NamedTagEventList list : tags) {
                 List<TagEvent> items = list.getItems();
-                if (items == null) {
+                if (items == null || items.isEmpty()) {
                     continue TAG_EVENT_LIST;
                 }
 
-                // latest item is the first
-                for (TagEvent item : items) {
-                    String image = item.getImage();
-                    if (Strings.isNotBlank(image)) {
-                        log.info("Found tag on ImageStream " + imageStreamName + " tag: " + image);
-                        return image;
-                    }
+                for (TagEvent tag : items) {
+                    latestTag = latestTag == null ? tag : newerTag(tag, latestTag);
                 }
             }
+
```

Comment:
It feels to me that  this block `if (latestTag != null ...)` should move outside the `for { }`  just like `currentImageStream`

Reply:
apparently, if we take this out of loop, logic will keep iterating till max iteration even after find the latest tag. Its better to return ith iteration after getting latest tag?